modules/rostests/apitests/user32/PrivateExtractIcons.c

```diff
@@ -167,3 +186,67 @@ START_TEST(PrivateExtractIcons)
         DeleteFileW(PathBuffer);
     }
 }
+
+static const struct tagPAIRSTESTS
+{
+    BYTE InCount;
+    BYTE UseHigh; // Extract a pair (high and low word sizes)
+    BYTE Library;
+    BYTE ReturnNT5;
+    BYTE CountNT5;
+    BYTE ReturnNT6;
+    BYTE CountNT6;
+} g_pairs[] =
+{
+    { 0, 0, 0,     1, 1, 1, 1 },
+    { 0, 0, 1,     0, 0, 0, 0 },
+    { 0, 1, 0,     1, 2, 1, 2 },
+    { 0, 1, 1,     0, 0, 0, 0 },
+    { 1, 0, 0,     1, 1, 1, 1 },
+    { 1, 0, 1,     1, 1, 1, 1 },
+    { 1, 1, 0,     1, 2, 1, 2 },
+    { 1, 1, 1,     2, 2, 0, 0 },
+    { 2, 0, 0,     1, 1, 1, 1 },
+    { 2, 0, 1,     2, 2, 2, 2 },
+    { 2, 1, 0,     1, 2, 1, 2 },
+    { 2, 1, 1,     2, 2, 2, 2 }, // This is the only way to extract a pair from a PE on NT6!
+    { 3, 0, 0,     1, 1, 1, 1 },
+    { 3, 0, 1,     3, 3, 3, 3 },
+    { 3, 1, 0,     1, 2, 1, 2 },
+    { 3, 1, 1,     4, 4, 0, 0 },
+    { 4, 0, 0,     1, 1, 1, 1 },
+    { 4, 0, 1,     4, 4, 4, 4 },
+    { 4, 1, 0,     1, 2, 1, 2 },
+    { 4, 1, 1,     4, 4, 4, 4 }
+};
+
+void TestPairExtraction(void)
+{
+    const HICON hInvalidIcon = (HICON)(INT_PTR)-2;
+    const BOOL IsNT6 = IsWindowsVistaOrGreater();
+    for (UINT i = 0; i < _countof(g_pairs); ++i)
+    {
+        UINT j, Count, ExpectedCount;
+        int RetVal, ExpectedRet;
+        UINT IcoSize = MAKELONG(32, g_pairs[i].UseHigh ? 16 : 0);
+        PCWSTR pszPath = g_pairs[i].Library ? L"%SystemRoot%\\system32\\shell32.dll" : L"%temp%\\sysicon.ico";
+        HICON hIcons[8];
+        for (j = 0; j < _countof(hIcons); ++j)
+            hIcons[j] = hInvalidIcon;
+
+        RetVal = PrivateExtractIconsW(pszPath, 0, IcoSize, IcoSize, hIcons, NULL, g_pairs[i].InCount, 0);
+        for (j = 0, Count = 0; j < _countof(hIcons); ++j)
+        {
+            if (hIcons[j] != hInvalidIcon && IsValidIcon(hIcons[j]))
+            {
+                DestroyIcon(hIcons[j]);
+                ++Count;
+            }
+        }
+
+        ExpectedRet = !IsNT6 ? g_pairs[i].ReturnNT5 : g_pairs[i].ReturnNT6;
+        ExpectedCount = !IsNT6 ? g_pairs[i].CountNT5 : g_pairs[i].CountNT6;
+        ok(RetVal == ExpectedRet, "RetVal must be %d for test %u but got %d\n", ExpectedRet, i, RetVal);
+        ok(Count == ExpectedCount, "Count must be %u for test %u but got %u\n", ExpectedCount, i, Count);
```

Comment:
To make these more easily parsable at the blink of an eye, I would suggest:
```suggestion
        ok(RetVal == ExpectedRet, "Test %u: RetVal must be %d but got %d\n", i, ExpectedRet, RetVal);
        ok(Count == ExpectedCount, "Test %u: Count must be %u but got %u\n", i, ExpectedCount, Count);
```